projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/secrets/DataJobsSecretsControllerTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2021-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/*
+ * Copyright 2021-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.secrets;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.vmware.taurus.exception.DataJobSecretsException;
+import com.vmware.taurus.exception.SecretStorageNotConfiguredException;
+import com.vmware.taurus.secrets.controller.DataJobsSecretsController;
+import com.vmware.taurus.secrets.controller.NoOpDataJobsSecretsController;
+import com.vmware.taurus.secrets.service.JobSecretsService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class DataJobsSecretsControllerTest {
+
+  @Mock private JobSecretsService jobSecretsService;
+
+  @InjectMocks private DataJobsSecretsController controller;
+  @InjectMocks private NoOpDataJobsSecretsController noOpsController;
+
+  @Test
+  void testDataJobSecretsUpdate() {
```

Comment:
Have you thought of those edge cases 
- Can job name be null ? (if yes we need a test)
- team name be null ? 
- Can the secrets map be null or empty ?
- Can the secrets map contain invalid characters  ?

Reply:
addressed.